src/Compilation.zig

```diff
@@ -700,7 +700,11 @@ fn addPackageTableToCacheHash(
                 const pkg_zig_file = try pkg.value.root_src_directory.join(allocator, &[_][]const u8{
                     pkg.value.root_src_path,
                 });
-                _ = try man.addFile(pkg_zig_file, null);
+                blk: {
+                    pkg.value.root_src_directory.handle.access(pkg.value.root_src_path, .{}) catch break :blk;
```

Comment:
I think this function should avoid file system access. I suggest to detect and store whether there is a build.zig file earlier, when constructing the package table.